pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -90,6 +90,17 @@ def __verify_train_loop_configuration(self, model: "pl.LightningModule") -> None
                 "(rather, they are called on every optimization step)."
             )
 
+        # Allow max_epochs or max_steps to be zero, since this will be handled by fit_loop.done
```

Comment:
is config validator the best place to put it? note this will delay validation until fit. But have we considered moving it directly into the FitLoop constructor? I don't see anything standing in the way of that, so I'm suggesting that for now.

Reply:
Moved into constructors